src/extensions/mountainview/MVSortingView/PreloadCheck.tsx

```diff
@@ -0,0 +1,115 @@
+import { Dialog } from '@material-ui/core';
+import React, { Fragment, FunctionComponent, useContext, useEffect, useMemo, useRef, useState } from 'react';
+import { createCalculationPool, HitherContext } from '../../common/hither';
+import { Recording, Sorting } from '../../extensionInterface';
+
+interface ChildProps {
+    preloadStatus?: 'waiting' | 'running' | 'finished'
+}
+
+type Props = {
+    sorting: Sorting
+    recording: Recording
+    children: React.ReactElement<ChildProps>
+}
+
+const calculationPool = createCalculationPool({maxSimultaneous: 6})
+
+const PreloadCheck: FunctionComponent<Props> = ({ recording, sorting, children }) => {
+    const hither = useContext(HitherContext)
+    const sortingObject = sorting.sortingObject
+    const recordingObject = recording.recordingObject
+    const [error, setError] = useState<Error | null>(null)
+    const [status, setStatus] = useState<'waiting' | 'running' | 'finished'>('waiting')
+    const [message, setMessage] = useState<string>('')
+    const runningState = useRef<{sortingObject: any, recordingObject: any}>({sortingObject: sorting.sortingObject, recordingObject: recording.recordingObject})
+
+    const matchesRunningState = useMemo(() => ((x: {recordingObject: any, sortingObject: any}) => (
+        (runningState.current.sortingObject === x.sortingObject) && (runningState.current.recordingObject === x.recordingObject)
+    )), [])
+
+    useEffect(() => {
+        if (status === 'waiting') {
+            runningState.current = {recordingObject, sortingObject}
```

Comment:
We need to keep track of the current recording/sorting objects so we can tell if they have been changed. If changed, we abandon the processing.